tests/integration/hsts/hsts_test.go

```diff
@@ -68,13 +68,13 @@ func (r *responseRecorder) Write(data []byte) (int, error) {
 func TestHSTSServeMuxInstall(t *testing.T) {
 	mux := safehttp.NewServeMux(&dispatcher{}, "foo.com")
 
+	mux.Install("hsts", hsts.Default())
+
```

Comment:
We have to do something to avoid users making the same mistake. I opened #117

Reply:
When we call `mux.Install` to install a plugin, we could check whether any handler was registered, in which case we panic because plugins should be installed before handlers in order to take effect. WDYT?

Reply:
sounds like a *meh* solution. We should discuss this with @kele 

Panicking on startup is fine but wouldn't it be better if we solved this on a type level?

Reply:
#117